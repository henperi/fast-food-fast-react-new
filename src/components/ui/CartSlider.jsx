import React from 'react';
import { connect } from 'react-redux';

export const CartSlider1 = (props) => {
  const { cart, slideCart } = props;
  return (
    <div className={`cart card-shadow text-center ${slideCart.slide && 'slide-from-right'}`}>
      <div className="right-nav close-button">
        <i className="fa fa-close" />
      </div>
      <br />
      <div className="container">
        My Cart
        {' '}
        <i className="fa fa-shopping-cart" />
      </div>
      <div className="card text-inverse">
        <span className="cart-count">
          {cart.length}
          {' '}
        </span>
        Items added
      </div>
      <div className="container">
        <a href="my-cart.html" className="btn btn-sm btn-green">
          View
        </a>
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  cart: state.cart,
  slideCart: state.slideCart,
});

export default connect(mapStateToProps)(CartSlider1);
