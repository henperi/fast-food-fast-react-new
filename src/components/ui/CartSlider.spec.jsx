import React from 'react';
import { shallow } from 'enzyme';
import { CartSlider1 } from './CartSlider';

test('should render CartSlider correctly', () => {
  const wrapper = shallow(<CartSlider1 cart={[]} slideCart={{ slide: false }} />);
  expect(wrapper).toMatchSnapshot();
});
