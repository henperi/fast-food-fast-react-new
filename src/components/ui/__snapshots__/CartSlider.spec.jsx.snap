// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CartSlider correctly 1`] = `
<div
  className="cart card-shadow text-center false"
>
  <div
    className="right-nav close-button"
  >
    <i
      className="fa fa-close"
    />
  </div>
  <br />
  <div
    className="container"
  >
    My Cart
     
    <i
      className="fa fa-shopping-cart"
    />
  </div>
  <div
    className="card text-inverse"
  >
    <span
      className="cart-count"
    >
      0
       
    </span>
    Items added
  </div>
  <div
    className="container"
  >
    <a
      className="btn btn-sm btn-green"
      href="my-cart.html"
    >
      View
    </a>
  </div>
</div>
`;
